baiji/test_config.py

```diff
@@ -1,28 +1,200 @@
 import unittest
-import os
-
+import tempfile
 from test.test_support import EnvironmentVarGuard
+from baiji.config import Settings
+from baiji.exceptions import AWSCredentialsMissing
+
+class FakeConfigFile(object):
+    def __init__(self, contents):
+        self.tf = tempfile.NamedTemporaryFile()
+        self.tf.write(contents)
+        self.tf.flush()
+        self.path = self.tf.name
+
+class FakeYAMLConfigFile(FakeConfigFile):
+    def __init__(self, contents):
+        import yaml
+        super(FakeYAMLConfigFile, self).__init__(yaml.dump(contents))
+
+class FakeConfConfigFile(FakeConfigFile):
+    def __init__(self, contents):
+        rendered_contents = ["[default]"] + ["{} = {}".format(k, v) for k, v in contents.items()]
+        super(FakeConfConfigFile, self).__init__("\n".join(rendered_contents))
+
+
+def AWSEnvironmentVarGuard():
+    env = EnvironmentVarGuard()
+    env.unset('AWS_ACCESS_KEY_ID')
+    env.unset('AWS_ACCESS_KEY')
+    env.unset('AWS_SECRET_ACCESS_KEY')
+    env.unset('AWS_SECRET')
+    env.unset('AWS_DEFAULT_REGION')
+    env.set('BODYLABS_CREDENTIAL_FILE', '/this_file_does_not_exist')
+    env.set('AWS_CONFIG_FILE', '/this_file_does_not_exist')
+    env.set('AWS_CREDENTIAL_FILE', '/this_file_does_not_exist')
+    return env
+
+class ConfigFileOverride(object):
+    def __init__(self):
+        self.env = AWSEnvironmentVarGuard()
+        self._dot_bodylabs = None
+        self._aws_credentials = None
+        self._aws_config = None
+    def __enter__(self):
+        return self.env.__enter__()
+    def __exit__(self, exception_type, exception_value, traceback):
+        return self.env.__exit__(exception_type, exception_value, traceback)
```

Comment:
They're temporary files:
```py
class FakeConfigFile(object):
    def __init__(self, contents):
        self.tf = tempfile.NamedTemporaryFile()
```
so when they go out of scope they're cleaned up automatically.